src/hyp_map.rs

```diff
@@ -144,6 +147,15 @@ const_assert!(PageSize::Size4k.is_aligned(UMODE_MAPPINGS_START));
 const UMODE_MAPPING_SLOTS: u64 = 2;
 // Maximum size of the private mappings area. Must be 4k-aligned.
 const UMODE_MAPPINGS_SIZE: u64 = UMODE_MAPPING_SLOTS * UMODE_MAPPING_SLOT_SIZE;
+const UMODE_MAPPINGS_END: u64 = UMODE_MAPPINGS_START + UMODE_MAPPINGS_SIZE;
+const_assert!(PageSize::Size4k.is_aligned(UMODE_MAPPINGS_END));
+
+/// Start of the U-mode Shared area.
+pub const UMODE_SHARED_START: u64 = UMODE_MAPPINGS_END + 4 * 1024 * 1024;
```

Comment:
bikeshed: Might `UMODE_INPUT` or `UMODE_INPUT_START` be a more precise name for this buffer? Sure it's "shared", but it's used solely for passing an input to the U-mode program.

Reply:
Yes. I am still keeping it open to the possibility of making this buffer read/writable. Name reflects that.